sensor/common/scannerclient/grpc_client.go

```diff
@@ -143,25 +143,20 @@ func dialV2() (ScannerClient, error) {
 
 // dialV4 connect to scanner V4 gRPC and return a new ScannerClient.
 func dialV4() (ScannerClient, error) {
-	endpoint, err := getScannerEndpoint(env.ScannerV4GRPCEndpoint)
+	ctx := context.Background()
+	c, err := client.NewGRPCScanner(ctx,
```

Comment:
Are both indexer and matcher going to be created here? Can you kindly remind me where do we set up the scannr mode?

Reply:
I think of the Scanner mode as an attribute of the Scanner deployment. You can deploy Scanner in matcher, indexer, or combo, currently based on attributes of the configuration file. Saying that, the clients should know the endpoints to each Scanner deployment that will provide the desired functionality (matching or indexing).

In the case of Sensor, we are looking to call only the Scanner Indexer, offered by `GetOrCreateImageIndex` method of the `scanner/pkg/client`. Under the hood, that will reach the endpoint defined in L148.

Reply:
So sounds like we are assuming the clients know what mode they are using and the according endpoint will be available. Is that right?

Reply:
The client decides which endpoint to connect to based on the configuration parameters. The different APIs will determine which Scanner backend or service is necessary.

Reply:
All right, assuming your "thumbs up" is a resolve.